### PR TITLE
flip/borders/rawprepare: distort_transform speedup

### PR DESCRIPTION
**Pull request by ralfbrown**
Found this branch still lying around....

Add early exit if there is no active distortion, plus some cleanup.  These iops don't have heavy-weight computation for determining the pixel distortion, so speedup is actually minimal.  But it brings them in line with other iops that have early exits.

Also skip parallelization if there are only a small number of points whose distortion needs to be computed.  The threshold could probably be increased quite a bit, given the relative overhead of thread management compared to the simple distortion arithmetic, but I was being conservative.

### CODE REVIEW

File: src/iop/borders.c [hunk 1]
```diff
@@ -220,10 +220,13 @@ int distort_transform(dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, floa
...
   const int border_size_l = border_tot_width * d->pos_h;
 
+  // nothing to be done if parameters are set to neutral values (no top/left border)
+  if (border_size_l == 0 && border_size_t == 0) return 1;
+
 #ifdef _OPENMP
 #pragma omp parallel for default(none) \
   dt_omp_firstprivate(points, points_count, border_size_l, border_size_t)  \
-  schedule(static)
+  schedule(static) if(points_count > 100)
```
Comment (TurboGit) [inline src/iop/borders.c:229]: This seems really wrong to me...

If OpenMP is active and we have less than 100 points we do nothing!

File: src/iop/borders.c [hunk 2]
```diff
@@ -243,10 +246,13 @@ int distort_backtransform(dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece,
...
```
Comment (TurboGit) [inline src/iop/borders.c:255]: likewise

**Review by TurboGit — CHANGES_REQUESTED** [review]
Probably something to rework. See comments.

File: src/iop/borders.c (hunk 1 above)
Comment (ralfbrown) [reply src/iop/borders.c:229]: No, the "if" on the OpenMP directive means only parallelize if the condition is true -- if it's false, run single-threaded.

OpenMP 4.5 specifications, section 2.5.1 (pages 50-51):
```
if (IfClauseValue = false)
   then number of threads = 1;
```

Comment (TurboGit) [reply src/iop/borders.c:229]: Oh ! That's part of the OpenMP directive, I was completely confused when seeing that. Sorry!

**Review by TurboGit — APPROVED** [review]
So finally code looks good!

**Pull request merged**